pkg/operator/controller/ingress/namespace.go

```diff
@@ -4,26 +4,128 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	corev1 "k8s.io/api/core/v1"
+
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 
 	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+	operatorcontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
 )
 
-// ensureRouterNamespace ensures all the necessary scaffolding exists for
-// routers generally, including a namespace and all RBAC setup.
-func (r *reconciler) ensureRouterNamespace() error {
-	ns := manifests.RouterNamespace()
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: ns.Name}, ns); err != nil {
-		if !errors.IsNotFound(err) {
-			return fmt.Errorf("failed to get router namespace %q: %v", ns.Name, err)
+// ensureRouterNamespace ensures that the router namespace exists.
+func (r *reconciler) ensureRouterNamespace() (bool, *corev1.Namespace, error) {
+	desired := manifests.RouterNamespace()
+
+	haveNamespace, current, err := r.currentRouterNamespace()
+	if err != nil {
+		return false, nil, err
+	}
+
+	switch {
+	case !haveNamespace:
+		if err := r.client.Create(context.TODO(), desired); err != nil {
+			return false, nil, fmt.Errorf("failed to create router namespace: %v", err)
+		}
+		log.Info("created router namespace", "desired", desired)
```

Comment:
We're inconsistent in this operator with respect to logging objects and their names.  This isn't a problem with this specific PR, but we should decide whether it is generally useful to log the entire object or better to log just the name.

Reply:
Right. I don't personally have a strong preference either way, although maybe we could get away with not logging entire object's during creation since we log diff's on updates?

Reply:
Yeah.  We can log interesting fields but otherwise avoid logging the entire object so as to reduce log noise.